### PR TITLE
fix detection of parallel mode

### PR DESCRIPTION
**Pull request by sam217pa**
Hi, 

I've been trying sfreemap, looks great, documentation is very clear.
I ran on an error when trying to reproduce the `map_posterior_distribution` part of the vignette:

```
Error in on_windows() : could not find function "on_windows"
```

I guess this function was superseded by `support_parallel_mode`.

This PR should fix it, along with what I supposed to be a minor logical error.

### CODE REVIEW

File: R/sfreemap.R [hunk 1]
```diff
@@ -9,7 +9,7 @@ sfreemap <- function(tree, tip_states, Q=NULL, type="standard", model="SYM", met
     # Should this program run in parallel?
     if (hasArg(parallel)) {
         parallel <- list(...)$parallel
-        if (all(parallel, !support_parallel_mode())) {
+        if (any(parallel, !support_parallel_mode())) {
```
Comment (dpasqualin) [inline R/sfreemap.R]: This is actually correct I think, I want to show a warning if the user requested the parallel mode but it is not available in the operating system. So it should be the same case as before:
```
if (isTRUE(parallel) && support_parallel_mode()) {
```

**Review by dpasqualin — CHANGES_REQUESTED** [review]

**Comment by dpasqualin** [comment]
Hi @sam217pa , thank you very much for your PR!!
I left a comment there, could you please check it and update the PR accordingly? Unfortunately I don't have time to test it now, but I would be happy to merge your PR this weekend and perhaps fix some of the other reported issues.

And please let me know if you manage to use the tool :)

**Comment by sam217pa** [comment]
my bad ! got it. I think [commit b96](https://github.com/dpasqualin/sfreemap/pull/18/commits/b96e0d73cd3de30de4073b3103ef539a30c86c27) fixes it.

I tested the parallelized function with the corals.trees dataset and it seems to work just fine on macOS, so second commit adds Darwin as list of supported OS.

**Review by dpasqualin — APPROVED** [review]

**Comment by dpasqualin** [comment]
Thank you very much for your feedback @sam217pa !!

**Pull request merged**